src/cli/api/api.ts

```diff
@@ -89,54 +92,103 @@ export async function fetchTrackingPlan(options: {
   email: string;
   APIVersion: string;
 }): Promise<RudderAPI.TrackingPlan> {
-  let url =
-    options.APIVersion === 'v1' ? `trackingplans/${options.id}` : `tracking-plans/${options.id}`;
-  url = options.version ? `${url}?version=${options.version}` : url;
-  const response = await apiGet<RudderAPI.GetTrackingPlanResponse>(
+  let response: RudderAPI.GetTrackingPlanResponse;
+  const { id, version, token, email } = options;
+
+  switch (options.APIVersion) {
+    case 'v1':
+      response = await _fetchTrackingPlanV1(id, version, token, email);
+      break;
+
+    case 'v2':
+      response = await _fetchTrackingPlanV2(id, version, token, email);
+      break;
+
+    default:
```

Comment:
From earlier changes, looks like default is v2.
is options.APIVersion optional ?

Reply:
I felt that if the version is whatever it would default to 2 and this might be wrong on our part of typer. We should fail and say we don't support this version. Wdyt ?

Reply:
who is setting up this apiversion ?

Reply:
apiVersion cannot be set apart from v1 and v2

Reply:
who sets this value? user or rudder-typer or ?

Reply:
Initially the typer sets it and afterwards the user can tweak the vales from .ruddertyper.yml file and we have restrictions enforced as what has to be set there